walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,151 @@
+        else { 
+            while (iter.hasNext()) {
+                String currentAttendee = (String)iter.next();
+                if (currEvent.getAttendees().contains(currentAttendee)) {
+                    eventsList.put(currEvent.getWhen().start(), currEvent);
+                    break;
+                }
+            }
+        }
+    }
+    
+    //Need this in order to iterate through the TreeMap
+    Set set = eventsList.entrySet();
+    
+    //finalTimes will contain all the times the meeting can happen
+    ArrayList<TimeRange> finalTimes = new ArrayList<TimeRange>();
+   
+    //If meeting duration > 24 hours, return no options
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        finalTimes.clear();
+    } 
+
+    //If request has no attendees or there are no events already set up, return whole day
+    else if (request.getAttendees().isEmpty() || events.isEmpty() || eventsList.isEmpty()) {
+        finalTimes.add(TimeRange.WHOLE_DAY);
+    }
+
+    int limitTime = TimeRange.START_OF_DAY;
+    boolean contained = false;
+    boolean overlapped = false;
+    Event nextEvent = null;
+
+    //Iterates through sorted important events to find the meeting times
+    for (Iterator iterator = set.iterator(); iterator.hasNext();) {
+        
+        Map.Entry mentry = (Map.Entry)iterator.next();
+        
+        //currentEvent is the event the loop is currently on
+        Event currentEvent = (Event)mentry.getValue(); 
+        
+        //nextEvent is the event that comes after the currentEvent in the TreeMap
+        if (iterator.hasNext()) { 
+            nextEvent = (Event)eventsList.higherEntry((int)mentry.getKey()).getValue();
+        }
+        
+        //in case two events overlap, and one contains the other, and there are no other events after those
+        if (contained && !iterator.hasNext()) {
+            //create new TimeRange
+            TimeRange newRange = TimeRange.fromStartEnd(limitTime, TimeRange.END_OF_DAY, true);
+            //if the duration of the new TimeRange is greater or equal to the duration of the meeting requested
+            //then this is a TimeRange we can use for the meeting to occur
+            if (newRange.duration() >= request.getDuration()) {
+                finalTimes.add(newRange);
+            }
+        }
+        //in case two events overlap but one does not contain the other, and there are no other events after those
+        else if (overlapped && !iterator.hasNext()) {
+            //create new TimeRange
+            TimeRange newRange = TimeRange.fromStartEnd(limitTime, TimeRange.END_OF_DAY, true);
+            //if the duration of the new TimeRange is greater or equal to the duration of the meeting requested
+            //then this is a TimeRange we can use for the meeting to occur
+            if (newRange.duration() >= request.getDuration()) {
+                finalTimes.add(newRange);
+            }
+        }
```

Comment:
Put them in the same if statement

`if ((contained || overlapped) && !iterator.hasNext()) { `